### PR TITLE
Play 2.7 upgrade

### PR DESCRIPTION
**Pull request by chandraAndem**

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -4,18 +4,16 @@ resolvers += "Typesafe Releases" at "https://repo.typesafe.com/typesafe/releases
 resolvers += MavenRepository("HMRC-open-artefacts-maven2", "https://open.artefacts.tax.service.gov.uk/maven2")
 resolvers += Resolver.url("HMRC-open-artefacts-ivy2", url("https://open.artefacts.tax.service.gov.uk/ivy2"))(Resolver.ivyStylePatterns)
 
-addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.0.0")
-
-addSbtPlugin("uk.gov.hmrc" % "sbt-git-versioning" % "2.1.0")
+addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.4.0")
```
Comment (MihailsC-hmrc) [inline project/plugins.sbt:7]: Optional suggestion: the latest is `3.5.0`

```suggestion
addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.5.0")
```

File: project/plugins.sbt [hunk 2]
```diff
@@ -4,18 +4,16 @@ resolvers += "Typesafe Releases" at "https://repo.typesafe.com/typesafe/releases
...
 
 addSbtPlugin("uk.gov.hmrc" % "sbt-artifactory" % "1.2.0")
```
Comment (MihailsC-hmrc) [inline project/plugins.sbt:9]: Optional cleanup: the `sbt-artifactory` is not needed and can be removed

File: conf/application.conf [hunk 3]
```diff
@@ -44,6 +44,7 @@ play.http.requestHandler = "uk.gov.hmrc.play.bootstrap.http.RequestHandler"
```
Comment (MihailsC-hmrc) [inline conf/application.conf:44]: I would suggest double check the microservice module we use from bootstrap library. Looks like the current `uk.gov.hmrc.play.bootstrap.MicroserviceModule` is deprecated in favour of a new `uk.gov.hmrc.play.bootstrap.backend.BackendModule`

**Review by MihailsC-hmrc — CHANGES_REQUESTED** [review]
In general, all looks fine. Just couple of things to double check.

File: conf/application.conf [hunk 4]
```diff
@@ -44,6 +44,7 @@ play.http.requestHandler = "uk.gov.hmrc.play.bootstrap.http.RequestHandler"
 
 # Global error handler
 play.http.errorHandler = "uk.gov.hmrc.play.bootstrap.http.JsonErrorHandler"
```
[the remark this answers is not in the document]
Comment (chandraAndem) [reply conf/application.conf]: Done

File: project/plugins.sbt (hunk 1 above)
Comment (chandraAndem) [reply project/plugins.sbt]: Done

File: project/plugins.sbt (hunk 2 above)
Comment (chandraAndem) [reply project/plugins.sbt]: Removed

File: app/config/MicroserviceAuthConnector.scala [hunk 5]
```diff
@@ -16,19 +16,16 @@
...
-class MicroserviceAuthConnector @Inject()( val environment: Environment,
-                                           val conf: Configuration,
-                                           val WSHttp: HttpClient,
-                                           ServicesConfig: ServicesConfig) extends PlayAuthConnector{
-  protected def runModeConfiguration: Configuration = conf
-  lazy val serviceUrl = ServicesConfig.baseUrl("auth")
-  lazy val http = WSHttp
-}
\ No newline at end of file
+class MicroserviceAuthConnector @Inject()(val conf: Configuration,
```
[the remark this answers is not in the document]
Comment (chandraAndem) [reply app/config/MicroserviceAuthConnector.scala]: Removed

File: conf/application.conf (hunk 3 above)
Comment (chandraAndem) [reply conf/application.conf:44]: Updated

**Review by MihailsC-hmrc — APPROVED** [review]

**Pull request merged**